plugins/jobs/src/js/JobCreateEditFormModal.tsx

```diff
@@ -0,0 +1,103 @@
+// tslint:disable:no-empty
+import * as React from "react";
+import { Observable } from "rxjs/Observable";
+import { BehaviorSubject } from "rxjs/BehaviorSubject";
+import { combineLatest } from "rxjs/observable/combineLatest";
+import "rxjs/add/operator/switchMap";
+
+import { componentFromStream } from "data-service";
+
+import MetronomeStore from "#SRC/js/stores/MetronomeStore";
+import Job from "#SRC/js/structs/Job";
+import {
+  METRONOME_JOB_CREATE_ERROR,
+  METRONOME_JOB_CREATE_SUCCESS,
+  METRONOME_JOB_UPDATE_ERROR,
+  METRONOME_JOB_UPDATE_SUCCESS
+} from "#SRC/js/constants/EventTypes";
+
+import JobFormModal, { ErrorMessage } from "./components/JobFormModal";
+
+interface JobCreateEditFormModalProps {
+  isEdit?: boolean;
+  job?: Job;
+  open: boolean;
+  onClose: () => void;
+}
+
+type IsJsonMode = boolean;
+
+type JobFormModalInput = [
+  JobCreateEditFormModalProps,
+  IsJsonMode,
+  ErrorMessage
+];
+
+const JobCreateEditFormModal = componentFromStream<JobCreateEditFormModalProps>(
+  props$ => {
+    const isJsonMode$ = new BehaviorSubject<IsJsonMode>(false);
+    const errorMessage$ = new BehaviorSubject<ErrorMessage>(null);
+
+    function handleInputModeToggle() {
+      isJsonMode$.next(!isJsonMode$.value);
+    }
+    function handleSubmit(isEdit: boolean, jobSpec: object, job?: Job) {
+      if (!isEdit) {
+        MetronomeStore.createJob(jobSpec);
+      } else {
+        MetronomeStore.updateJob(job && job.getId(), jobSpec);
+      }
+    }
+    function handleErrorMessageChange(errorMessage: ErrorMessage) {
+      errorMessage$.next(errorMessage);
+    }
+    function metronomeSuccessHandler() {
+      console.log("close"); // tslint:disable-line
+      // props.onClose(); // Props aren't there yet to use ¯\_(ツ)_/¯
+    }
+    MetronomeStore.addListener(
+      METRONOME_JOB_CREATE_ERROR,
+      handleErrorMessageChange
+    );
+    MetronomeStore.addListener(
+      METRONOME_JOB_UPDATE_ERROR,
+      handleErrorMessageChange
+    );
+    MetronomeStore.addListener(
+      METRONOME_JOB_CREATE_SUCCESS,
+      metronomeSuccessHandler
+    );
+    MetronomeStore.addListener(
+      METRONOME_JOB_UPDATE_SUCCESS,
+      metronomeSuccessHandler
+    );
+
+    const state = [isJsonMode$, errorMessage$];
+    return combineLatest<JobFormModalInput>([props$, ...state])
```

Comment:
interesting, I had to go with `Subscribable<React.ReactNode>` 🤔 but the way i declared types is a bit different